llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -1496,12 +1496,18 @@ class TargetInstrInfo : public MCInstrInfo {
   /// to TargetPassConfig::createMachineScheduler() to have an effect.
   ///
   /// \p BaseOps1 and \p BaseOps2 are memory operands of two memory operations.
+  /// \p Offset1 and \p Offset2 are the byte offsets for the memory
```

Comment:
Looking at ElementCount, I'm not sure it's semantically the right abstraction? e.g. for `LW %0:gpr, 12` we'd be passing `Offset1=12` and `OffsetIsScalable1=false`. `ElementCount` doesn't seem right for representing that.

Reply:
The docs for ElementCount are a bit confusingly worded, but I think this is the right abstraction here.  An Offset1=12 and OffsetIsScalable1=false would be a ElementCount::getFixed(12).

Reply:
Is ElementCount able to represent a negative offset? The current signature uses int64_t.

Reply:
That is a good point.  I don't think ElementCount can be negative.  We could either introduce a new subclass of FixedOrScalableQuantity, or just go with what @asb proposed.

Reply:
Oh right - we do not have an abstraction for negative scalable values. It would be ideal though, I can see `OffsetInBytes` derived from `FixedOrScalableQuantity` being useful in other places. I won't hold your work here - please go ahead with your code, we can fix it later if needed.

Reply:
Thank you all, and sorry for not understanding the suggestion at first Francesco. The docs for ElementCount indeed weren't clear to me.

I've added a FIXME note that moving to a new abstraction would be better. Would you be happy for me to land this PR as-is? I'm not opposed to looking to introduce this new abstraction, but if that's your desired route would appreciate some pointers to where else it might be used.